INSTALL.md

```diff
@@ -87,3 +87,102 @@ environment using:
 
     conda deactivate
     conda env remove -n compiler_gym
+
+## Building from source with CMake
+
+### Dependency instructions for Ubuntu
+
+#### CMake
+Requires CMake (==3.20).
+```
+wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | sudo tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
+sudo apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main"
+sudo apt update
+sudo apt install kitware-archive-keyring
+sudo rm /etc/apt/trusted.gpg.d/kitware.gpg
+sudo apt update
+sudo apt install cmake=3.20\* cmake-data=3.20\*
+```
+
+#### Compiler
+
+```
+sudo apt-get install clang-9 clang++-9 lld-9 patchelf tar bzip2
+```
+
+#### Ninja
+```
+sudo apt-get install ninja-build
+```
+
+#### Python
+```
+sudo apt-get install python3 python3-pip python3-venv
+```
+
+Setup a virtual environment.
+```
+python3 -m venv compiler_gym_venv
+source compiler_gym_venv/bin/activate
+```
+
+```
+pip install --upgrade pip
+# in the source directory
+python -m pip install -r requirements.txt
+```
+
+#### programl
+To build programl internally it requires Bazel.
+```
+sudo apt install apt-transport-https curl gnupg
+curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg
+sudo mv bazel.gpg /etc/apt/trusted.gpg.d/
+echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
+sudo apt update && sudo apt install bazel-4.0.0
+```
+
+#### protobuf
+To build protobuf internally:
+```
+sudo apt-get install autoconf libtool make
+```
+
+### Build speed optimization
+
+```
+# For faster rebuilds
+sudo apt-get install ccacke
+```
+
+### Dependency Arguments
+By default most dependencies are built together with Compiler Gym. To search for a dependency instead use:
+```
+-DCOMPILER_GYM_<dependency>_PROVIDER=external
+```
+
+* `COMPILER_GYM_BOOST_PROVIDER`
+* `COMPILER_GYM_GFLAGS_PROVIDER`
+* `COMPILER_GYM_GLOG_PROVIDER`
+* `COMPILER_GYM_GRPC_PROVIDER`
+* `COMPILER_GYM_GTEST_PROVIDER`
+* `COMPILER_GYM_NLOHMANN_JSON_PROVIDER`
+* `COMPILER_GYM_PROTOBUF_PROVIDER`
+
+```
+export SOURCE_DIR=<path to source directory>
+export BOULD_DIR=<path to build directory>
+
+cmake -GNinja \
+  -DCMAKE_C_COMPILER=clang-9 \
+  -DCMAKE_CXX_COMPILER=clang++-9 \
+  -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \ # For faster rebuilds, can be removed
+  -DCMAKE_EXE_LINKER_FLAGS_INIT="-fuse-ld=lld" -DCMAKE_MODULE_LINKER_FLAGS_INIT="-fuse-ld=lld" -DCMAKE_SHARED_LINKER_FLAGS_INIT="-fuse-ld=lld" \ # For faster builds, can be removed
```

Comment:
I reorganized the instructions and this problem is not present there.

Reply:
Looks like this optional arguments are still in the base command though. Tbh I think it would be fine to simply update the `lld` to `lld-9` and then include them as part of the base command, since the `lld-9` package is part of the required dependencies.

Reply:
I updated to use `lld-9` explicitly. I think on your system, `lld-9` is not the default `lld`.